Code/Source/Frame/NamespaceConfiguration.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+* Copyright (c) Contributors to the Open 3D Engine Project.
+* For complete copyright and license terms please see the LICENSE at the root of this distribution.
+*
+* SPDX-License-Identifier: Apache-2.0 OR MIT
+*
+*/
+
+#pragma once
+
+#include "Frame/NamespaceConfiguration.h"
+#include "Utilities/ROS2Names.h"
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/EditContextConstants.inl>
+
+namespace ROS2
+{
+    void NamespaceConfiguration::PopulateNamespace(bool isRoot, const AZStd::string &entityName)
+    {
+        m_isRoot = isRoot;
+        m_entityName = entityName;
+        OnNamespaceStrategySelected();
+    }
+
+    void NamespaceConfiguration::UpdateNamespace()
+    {
+        auto nss = m_namespaceStrategy;
+        if (NamespaceStrategy::Custom == nss)
+        {   // Leave the name as it is
+            return;
+        }
+
+        if (NamespaceStrategy::Empty == nss || (!m_isRoot && NamespaceStrategy::Default == nss))
+        {
+            m_namespace = "";
+        }
+        else if (NamespaceStrategy::FromEntityName == nss || (m_isRoot && NamespaceStrategy::Default == nss))
+        {
+            m_namespace = ROS2Names::RosifyName(m_entityName);
+        }
```

Comment:
Suggestion: For readability, can we 'switch' this to a switch statement?
```suggestion
        switch(m_namespaceStrategy)
        {
            case NamespaceStrategy::Empty:
                m_namespace = "";
                break;
            case NamespaceStrategy::Default:
                m_namespace = ROS2Names::RosifyName(m_entityName) ? m_isRoot : "";
                break;
            case NamespaceStrategy::FromEntity:
                m_namespace = ROS2Names::RosifyName(m_entityName);
                break;
            case NamespaceStrategy::Custom:
                // Leave the namespace as is
                break;
             default:
                AZ_Assert(false, "Unhandled namespace strategy");
                break;
        }
```

Reply:
I decided against a switch here since I thought there will be no more strategies to add and one early return + one if/else looks better. But I can change it, no strong opinion here, just a bit of dislike for switch.

Reply:
I went with it, only switched: `m_namespace = ROS2Names::RosifyName(m_entityName) ? m_isRoot : "";`
to `m_namespace = m_isRoot ? ROS2Names::RosifyName(m_entityName) : "";`
and FromEntity -> FromEntityName

Reply:
Thanks. IMHO, the switches on enumerations makes it easier to read the logic flow in these situations.